base/math.jl

```diff
@@ -536,38 +536,56 @@ Stacktrace:
 [...]
 ```
 """
-hypot(x::Number, y::Number) = hypot(promote(x, y)...)
-function hypot(x::T, y::T) where T<:Number
-    ax = abs(x)
-    ay = abs(y)
-    if ax < ay
-        ax, ay = ay, ax
-    end
+function hypot(x::Number, y::Number)
+	#= Keep specialized two argument version, because it is faster
+	than generic varargs version below =#
+	ay, ax = minmax(abs(x), abs(y))
     if iszero(ax)
         r = ay / oneunit(ax)
     else
         r = ay / ax
     end
-
     rr = ax * sqrt(1 + r * r)
 
-    # Use type of rr to make sure that return type is the same for
-    # all branches
-    if isnan(r)
-        isinf(ax) && return oftype(rr, Inf)
-        isinf(ay) && return oftype(rr, Inf)
-        return oftype(rr, r)
-    else
-        return rr
-    end
+	if isnan(r)
+		if isinf(x) || isinf(y)
+			return oftype(rr, Inf)
+		else
+			return oftype(rr, NaN)
+		end
+   	else
+   		return rr
+   	end
```

Comment:
`oftype(rr, r)` seems better here, so that we don't have to construct a new `NaN` value.  (Not all `NaN` values are the same.)

Reply:
Done, thanks again!